common/achievement/create.ts

```diff
@@ -119,7 +119,7 @@ async function createNextUserAchievement<ID extends ActionID>(
         });
         metrics.AchievementsCreated.inc({
             id: createdUserAchievement.template.id.toString(),
-            title: createdUserAchievement.template.title,
+            uid: getAchievementTemplateUID(createdUserAchievement.template),
```

Comment:
I would say that it's a real UID. The normal ID can change depending on the order of insertion, but the combination of group & groupOrder is the only way to identify a specific achievement right now. 
Theoretically, we could even create unique index in the DB.

Reply:
Then maybe we should drop the template.id, and create a primary key on (group, group_order) instead ...

Reply:
What's the advantage?

Reply:
Proper relational modeling :) - Apart from that none

Reply:
What about increased complexity? Instead of working with number IDs, like for every other model in our system, we would have to always pass these two variables. Furthermore, the connection to user_achievements will be more complex.

Do you have any major downsides in mind? You have much more experience working with Databases than I have.

Reply:
Fair enough, code complexity is a point. And yeah, a high cardinality composite foreign key could be a problem for Postgres (not so much for other DBs)